src/uct/sm/cma/cma_ep.c

```diff
@@ -55,67 +55,52 @@ ucs_status_t uct_cma_ep_common_zcopy(uct_ep_h tl_ep,
                                                      unsigned long),
                                      char *fn_name)
 {
+    uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
+    void *remote_ptr = UCS_PTR_BYTE_OFFSET(remote_addr, 0);
+    size_t iov_idx   = 0;
+    size_t local_iov_idx;
     ssize_t ret;
-    ssize_t delivered = 0;
-    size_t iov_it;
-    size_t iov_it_length;
-    size_t iov_slice_length;
-    size_t iov_slice_delivered;
-    size_t local_iov_it;
-    size_t length = 0;
+    size_t local_iov_cnt;
+    size_t length;
+    size_t cur_iov_cnt;
     struct iovec local_iov[UCT_SM_MAX_IOV];
     struct iovec remote_iov;
-    uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
-
-    do {
-        iov_it_length = 0;
-        local_iov_it = 0;
-        for (iov_it = 0; iov_it < ucs_min(UCT_SM_MAX_IOV, iovcnt); ++iov_it) {
-            iov_slice_delivered = 0;
 
-            /* Get length of the particular iov element */
-            iov_slice_length = uct_iov_get_length(iov + iov_it);
+    while (iovcnt - iov_idx) {
+        cur_iov_cnt   = ucs_min(iovcnt - iov_idx, UCT_SM_MAX_IOV);
```

Comment:
it looks like CMA is trying to support iovcnt>UCT_SM_MAX_IOV, even though it doesn't have to according to UCT API
can you pls take this opportunity to simplify the code and remove the external while loop?

Reply:
CMA reports that it supports maximum available IOVs
https://github.com/openucx/ucx/blob/f2a2cb3c0dac90649781a0f9edcfa78562298186/src/uct/sm/cma/cma_iface.c#L42